### PR TITLE
Set annotation values readonly, if only attach points contain dual attach point without source

### LINKED ISSUE #26408: Making annotation values readonly has being commented out in JDK11 migration and needs to be re-enabled
**Description:**
https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java#L1086

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by chiranSachintha**
Fixes #26408 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 1]
```diff
@@ -1081,10 +1082,12 @@ public void visit(BLangAnnotation annotationNode) {
...
-        // TODO: need to check this. Balo creation in java module fails in java11 migration.
-//        if (annAttachmentNode.expr != null) {
-//            annAttachmentNode.expr = visitCloneReadonly(annAttachmentNode.expr, annAttachmentNode.expr.type);
-//        }
+        for (AttachPoint point : annAttachmentNode.annotationSymbol.points) {
+            if (annAttachmentNode.expr != null && !point.source) {
+                annAttachmentNode.expr = visitCloneReadonly(annAttachmentNode.expr, annAttachmentNode.expr.type);
+                break;
+            }
+        }
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:1090]: We don't need to iterate through the points if `annAttachmentNode.expr` is null, right? 

Shall we refactor this to check that first?

```suggestion
        if (annAttachmentNode.expr != null) {
            for (AttachPoint point : annAttachmentNode.annotationSymbol.points) {
                if (!point.source) {
                    annAttachmentNode.expr = visitCloneReadonly(annAttachmentNode.expr, annAttachmentNode.expr.type);
                    break;
                }
            }
        }
```

Comment (chiranSachintha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java]: Yes. Changed.

**Review by MaryamZi — APPROVED** [review]

**Comment by MaryamZi** [comment]
@chiranSachintha, tests related to this will be enabled separately, right?

**Comment by chiranSachintha** [comment]
> @chiranSachintha, tests related to this will be enabled separately, right?
Yes @MaryamZi . With this PR(https://github.com/ballerina-platform/ballerina-lang/pull/26395)

**Pull request merged**